hookman/hook_man_generator.py

```diff
@@ -0,0 +1,265 @@
+        list_with_private_members = []
+        from textwrap import dedent
+
+        file_content += dedent(f"""\
+        #include <functional>
+
+        namespace HookMan {{
+
+        template <typename F_TYPE> std::function<F_TYPE> register_c_func(uintptr_t p) {{
+        {INDENTATION}return std::function<F_TYPE>(reinterpret_cast<F_TYPE *>(p));
+        }}
+
+        class HookCaller {{
+        public:
+        """)
+
+        list_with_hook_calls += [
+            f'{INDENTATION}inline {hook.r_type} {hook.name}({hook.args_with_type}) {{ return this->_{hook.name}({hook.args}); }}' + NEW_LINE
+            for hook in self.hooks
+        ]
+
+        list_with_private_members += [
+            f'{INDENTATION}std::function<{hook.r_type}({hook.args_type})> _{hook.name};' + NEW_LINE
+            for hook in self.hooks
+        ]
+
+        list_with_set_functions += [
+            f'{1*INDENTATION}void set_{hook.name}_function(uintptr_t pointer) {{' + NEW_LINE +
+            f'{2*INDENTATION}this->_{hook.name} = register_c_func<{hook.r_type}({hook.args_type})>(pointer);' + NEW_LINE +
+            f'{1*INDENTATION}}}' + 2 * NEW_LINE
+            for hook in self.hooks
+        ]
+        file_content += list_with_hook_calls
+        file_content += NEW_LINE
+        file_content += list_with_set_functions
+        file_content += "private:" + NEW_LINE
+        file_content += list_with_private_members
+        file_content += "};" + NEW_LINE + "}" + NEW_LINE
+
+        return file_content
+
+    def _hook_caller_python_content(self) -> List[str]:
+        """
+        Create a .cpp file to bind python and cpp code with PyBind11
+        """
+        file_content = []
+        file_content += dedent(f"""\
+            #include <pybind11/pybind11.h>
+            #include <hookcaller.hpp>
+
+            namespace py = pybind11;
+
+            PYBIND11_MODULE({self.pyd_name}, m) {{
+                py::class_<HookMan::HookCaller>(m, "HookCaller")
+                    .def(py::init<>())
+        """)
+        file_content += [
+            f'{2*INDENTATION}.def("set_{hook.name}_function", &HookMan::HookCaller::set_{hook.name}_function)' + NEW_LINE
+            for hook in self.hooks
+        ]
+        file_content += f'{2*INDENTATION};' + NEW_LINE + '}' + NEW_LINE
+        return file_content
+
+    def _generate_cmake_files(self, dst_path: Path):
+        hook_caller_hpp = Path(dst_path / 'cpp' / 'CMakeLists.txt')
+        hook_caller_python = Path(dst_path / 'binding' / 'CMakeLists.txt')
+        from textwrap import dedent
+
+        with open(hook_caller_hpp, mode='w') as file:
+            file.writelines(dedent(f"""\
+                add_library({self.pyd_name}_interface INTERFACE)
+                target_include_directories({self.pyd_name}_interface INTERFACE ./)
+                """))
+
+        with open(hook_caller_python, mode='w') as file:
+            file.writelines(dedent(f"""\
+            include(pybind11Tools)
+

```

Comment:
Mixed line-endings

Reply:
Commit https://github.com/ESSS/hookman/pull/1/commits/b2980dd1efe4669c3bb69b739a065c3f5f17997e normalize the line endings